test/Formik.test.tsx

```diff
@@ -470,13 +481,27 @@ describe('<Formik>', () => {
         await wait(() => expect(onSubmit).toBeCalled());
       });
 
-      it('should not submit the form if invalid', () => {
+      it('should not submit the form if invalid', async () => {
+        const onSubmit = jest.fn();
+        const validate = jest.fn(() => Promise.reject({ name: 'Error!' }));
+        const { getByTestId } = renderFormik({ onSubmit, validate });
+
+        fireEvent.submit(getByTestId('form'));
+        await wait(() => expect(onSubmit).not.toBeCalled());
+      });
+
+      it('should not submit the form if validate function rejects with an error', async () => {
         const onSubmit = jest.fn();
-        const validate = jest.fn(() => Promise.resolve({ name: 'Error!' }));
+        const validate = jest.fn(
+          () =>
+            new Promise(() => {
+              throw new Error('Oops');
```

Comment:
This test covers the case described here: https://github.com/jaredpalmer/formik/pull/1201/files#r243037849